runtime/vm/classloadersearch.c

```diff
@@ -237,3 +241,81 @@ addZipToLoader(J9JavaVM * vm, const char * filename, J9ClassLoader * classLoader
 
 	return rc;
 }
+
+jclass
+getJimModules(J9VMThread *currentThread)
+{
+	J9JavaVM *vm = NULL;
+	vm = currentThread->javaVM;
+	if (NULL == vm->jimModules) {
+		const J9InternalVMFunctions *vmFuncs = NULL;
+		/* no need for mutual exclusion since repeating this is harmless */
+		J9Class *j9JimModules = NULL;
+		internalEnterVMFromJNI(currentThread);
+		vmFuncs = vm->internalVMFunctions;
+		j9JimModules = vmFuncs->internalFindKnownClass(currentThread,
+				J9VMCONSTANTPOOL_JDKINTERNALMODULEMODULES,
+				J9_FINDKNOWNCLASS_FLAG_INITIALIZE | J9_FINDKNOWNCLASS_FLAG_NON_FATAL);
+		if (NULL != j9JimModules) {
+			/*
+			 * The address of j9class->classObject is considered to be a JNI global reference.
+			 * Note that the class in question is not a candidate for unloading.
+			 */
+			vm->jimModules = (jclass)&j9JimModules->classObject;
+		} else {
+			currentThread->currentException = NULL;
+			currentThread->privateFlags &= ~J9_PRIVATE_FLAGS_REPORT_EXCEPTION_THROW;
+		}
+		internalExitVMToJNI(currentThread);
+	}
+	return vm->jimModules;
+}
+
+/**
+ * Loads the named module using jdk.internal.module.Modules.loadModule()
+ *
+ * @param [in] vm pointer to JVM
+ * @param [in] moduleName name of module to load
+ * @param [out] modulePtr pointer to return module object.  May be null.
+ *
+ * @return CLS_ERROR_NONE on success, other value on failure
+ */
+static UDATA
+loadModule(J9JavaVM *vm, const char *moduleName, jobject *modulePtr)
+{
+	UDATA rc = CLS_ERROR_NONE;
+	J9VMThread *currentThread = currentVMThread(vm);
```

Comment:
Where this function is used, we've already retrieved the current thread: it would make sense to avoid redoing that by passing `currentThread` or `env` instead of `vm`.

Reply:
Will fix in https://github.com/eclipse/openj9/issues/2437